test/Cardano/CoinSelectionSpec.hs

```diff
@@ -146,21 +150,17 @@ data CoinSelectionResult = CoinSelectionResult
 -- | Generate a 'UTxO' and 'TxOut' matching the given 'Fixture', and perform
 -- the given coin selection on it.
 coinSelectionUnitTest
-    :: ( CoinSelectionOptions ErrValidation
-         -> NonEmpty TxOut
-         -> UTxO
-         -> ExceptT (ErrCoinSelection ErrValidation) IO (CoinSelection, UTxO)
-       )
+    :: CoinSelectionAlgorithm IO ErrValidation
     -> String
     -> Either (ErrCoinSelection ErrValidation) CoinSelectionResult
     -> CoinSelectionFixture
     -> SpecWith ()
```

Comment:
Using the `CoinSelectionAlgorithm` type here makes it clear that this function will work on any coin selection algorithm.